stake-pool/program/src/state.rs

```diff
@@ -113,16 +113,26 @@ impl StakePool {
         )
         .ok()
     }
-    /// calculate the fee in pool tokens that goes to the manager
+
+    /// Calculate the fee in pool tokens that goes to the manager
+    ///
+    /// This function assumes that `reward_lamports` has not already been added
+    /// to the stake pool's `total_stake_lamports`
     pub fn calc_fee_amount(&self, reward_lamports: u64) -> Option<u64> {
         if self.fee.denominator == 0 {
             return Some(0);
         }
-        let pool_amount = self.calc_pool_tokens_for_deposit(reward_lamports)?;
+        let total_stake_lamports = self.total_stake_lamports.checked_add(reward_lamports)? as u128;
+        let fee_lamports = (reward_lamports as u128)
+            .checked_mul(self.fee.numerator as u128)
+            .unwrap()
```

Comment:
This multiplication is guaranteed not to overflow, because `rewards_lamports` and `fee.numerator` are both `u64`s that get widened to `u128`, and `u64::MAX * u64::MAX < u128::MAX`.